src/main/java/io/managed/services/test/dataplane/FleetshardUtils.java

```diff
@@ -0,0 +1,71 @@
+package io.managed.services.test.dataplane;
+
+import io.fabric8.kubernetes.api.model.KubernetesResourceList;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.dsl.MixedOperation;
+import io.fabric8.kubernetes.client.dsl.Resource;
+import io.fabric8.openshift.client.OpenShiftClient;
+import io.managed.services.test.k8.managedkafka.v1alpha1.ManagedKafka;
+import io.managed.services.test.k8.managedkafka.v1alpha1.ManagedKafkaAgent;
+import lombok.extern.log4j.Log4j2;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+@Log4j2
+public class FleetshardUtils {
+
+    public static MixedOperation<ManagedKafka, KubernetesResourceList<ManagedKafka>, Resource<ManagedKafka>> managedKafka(KubernetesClient client) {
+        return client.resources(ManagedKafka.class);
+    }
+
+    public static MixedOperation<ManagedKafkaAgent, KubernetesResourceList<ManagedKafkaAgent>, Resource<ManagedKafkaAgent>> managedKafkaAgent(KubernetesClient client) {
+        return client.resources(ManagedKafkaAgent.class);
+    }
+
+    public static int getCountOfExistingGivenManagedKafkaCRType(OpenShiftClient oc, ManagedKafkaType mkType) {
```

Comment:
I had another look at this, and I think it is not taking onto account the streaming unit notion. It assumes that each Mk CR represents 1 streaming unit. This is not always true as one MK CR could be 2 streaming units. The streaming unit consumption value of a CR is indicated by this https://github.com/bf2fc6cc711aee1a0c2a/e2e-test-suite/blob/04c43345024f9dff671221c74a9422b6f5360ca2/src/main/java/io/managed/services/test/k8/managedkafka/v1alpha1/ManagedKafka.java#L39 label.

So the streaming unit count of an MK CR is actually given by the value of that label.
And the total sum for all the CRs represents the count.

A suggestion (not type from IDE, so the syntax could be off).
```
return (int) FleetshardUtils.managedKafka(oc).inAnyNamespace().list().getItems()
            .stream()
            .filter(e -> mkType.toString().equals(e.getLabel("bf2.org/kafkaInstanceProfileType").orElse("notPresent")))
            .map(e -> Integer.valueOf(e.getLabel(PROFILE_QUOTA_CONSUMED).orElse("1")))
            .sum()
```

Reply:
yeah this is great suggestion.